### DISCUSSION TITLE
New in v8.12.0 - Improved yanking and toggling of system prompts

### DISCUSSION THREAD

**Discussion by olimorris** [olimorris/codecompanion.nvim#371] [Announcements]
A big shout out to @antosha417 for his work in #365. We now have the ability to visually see yanked code blocks in the chat buffer with `gy` as well as being able to yank any code block that the cursor is in. This builds on the yanking of the last code block that was added in `v8.7.0`.

New in `v8.12.0` is the ability to toggle on/off the default system prompt. My preferred method of engaging with LLMs is through CodeCompanion and being able to hit `gs` and toggle it off and have a conversation is much more convenient.

**Comment by bassamsdata** [comment]
Thank you for this `gs` feature. I always use other apps for non-code dialogs with LLMs since Claude was pretty unhappy with non-code conversations due to the system prompt. This is a golden feature.

**Reply by bassamsdata** [reply]
I was thinking of whether it's possible to adjust the temperature when no system prompt is active, especially for more creative/out-of-the-box responses :) I’m not sure how feasible it is to implement since the temperature setting is defined individually in each adapter.

**Reply by olimorris** [reply]
> I was thinking of whether it's possible to adjust the temperature when no system prompt is active, especially for more creative/out-of-the-box responses :) I’m not sure how feasible it is to implement since the temperature setting is defined individually in each adapter.

There is a setting for this. Show settings I believe it's called

**Reply by bassamsdata** [reply]
ops, I didn't realize that "show_settings" was for changing settings too; I thought it was just for displaying them. Thanks for clarifying. The only issue I think enabling this feature result in losing the `ga` one.
